### ISSUE TITLE
Preview links broken in SS 4.8

### ISSUE THREAD

**Issue by tardinha** [silverstripe-labs/silverstripe-sharedraftcontent#143]
We just upgraded to SS 4.8 are errors on /preview links

[2021-04-15 09:39:48] error-log.WARNING: E_WARNING: session_start(): The session id is too long or contains illegal characters, valid characters are a-z, A-Z, 0-9 and '-,' {"code":2,"message":"session_start(): The session id is too long or contains illegal characters, valid characters are a-z, A-Z, 0-9 and '-,'","file":"vendor/silverstripe/framework/src/Control/Session.php","line":330} []

[2021-04-15 09:39:48] error-log.WARNING: E_WARNING: session_start(): Failed to read session data: files (path: /var/lib/php/sessions) {"code":2,"message":"session_start(): Failed to read session data: files (path: /var/lib/php/sessions)","file":"vendor/silverstripe/framework/src/Control/Session.php","line":330} []

### PRs
- [x] https://github.com/silverstripe/silverstripe-sharedraftcontent/pull/144

**Comment by tardinha** [comment]
Looks like the issue is with version 2.4.0 of this module, not an issue with SS4.8 ;)

Confirmed SS4.8 plus 2.3.5 work as expected.

**Comment by wilr** [comment]
@tardinha By any chance are you using the Trailing Slash middleware (https://addons.silverstripe.org/add-ons/axllent/silverstripe-trailing-slash). I've noticed the same issue, it's caused by the new `getRenderedPageByURLSegment` method

**Comment by tardinha** [comment]
@wilr yep!!

**Comment by wilr** [comment]
@tardinha I've pushed up a fix to this module and updated the trailing slash version (https://github.com/axllent/silverstripe-trailing-slash/pull/14) if you want to give those 2 forks a go.

**Comment by emteknetnz** [comment]
We've merged and tagged a pull-request that should have fixed this https://github.com/silverstripe/silverstripe-sharedraftcontent/releases/tag/2.4.1

I've mark this issue as closed

Could you please confirm that everything is working from your end with both this new patch of this and the latest version axllent trailing slash.  If not then please reply on this issue

Cheers

**Comment by normann** [comment]
> Could you please confirm that everything is working from your end with both this new patch of this and the latest version axllent trailing slash. If not then please reply on this issue

Hi @emteknetnz,

I happened to have two projects that install both `sharedraftcontent` (v2.4.0) and `silverstripe-trailing-slash` (v2.2.5) modules, and recently updated the 2 sites to  4.8, and the preview link didn't work after the upgrade
Following this issue and fixes in the post, then I updated `sharedraftcontent` to 2.4.2 on one project, and to 2.4.1 on another project, just for testing (both versions contain this patch). Unfortunately, none of them works. 
Then I made further testing on one project by downgrading `recipe-collaboration from 1.8.0 to 1.7.1, which will downgrade `sharedraftcontent`from 2.4.0 to 2.3.x-dev. Then, the preview link began to work.
